We don't want the cairo dependency. Unfortunately simply checking whether its present 
isn't good enough. If its not in DEPENDS, it can disappear half way through building.
We therefore need to explictly disable it.

RP
2012/10/5

Index: libdrm-2.4.39/configure.ac
===================================================================
--- libdrm-2.4.39.orig/configure.ac	2012-08-24 14:54:42.000000000 +0000
+++ libdrm-2.4.39/configure.ac	2012-10-05 11:37:52.484821221 +0000
@@ -63,6 +63,11 @@
 	      [Disable KMS mm abstraction library (default: auto)]),
 	      [LIBKMS=$enableval], [LIBKMS=auto])
 
+AC_ARG_ENABLE(cairo,
+	      AS_HELP_STRING([--disable-cairo],
+	      [Disable cairo (default: auto)]),
+	      [ENABLECAIRO=$enableval], [ENABLECAIRO=auto])
+
 AC_ARG_ENABLE(intel,
 	      AS_HELP_STRING([--disable-intel],
 	      [Enable support for intel's KMS API (default: auto)]),
@@ -201,9 +206,12 @@
 	AC_DEFINE(HAVE_EXYNOS, 1, [Have EXYNOS support])
 fi
 
-PKG_CHECK_MODULES(CAIRO, cairo, [HAVE_CAIRO=yes], [HAVE_CAIRO=no])
-if test "x$HAVE_CAIRO" = xyes; then
-	AC_DEFINE(HAVE_CAIRO, 1, [Have cairo support])
+HAVE_CAIRO=no
+if test "x$ENABLECAIRO" = xyes; then
+	PKG_CHECK_MODULES(CAIRO, cairo, [HAVE_CAIRO=yes], [HAVE_CAIRO=no])
+	if test "x$HAVE_CAIRO" = xyes; then
+		AC_DEFINE(HAVE_CAIRO, 1, [Have cairo support])
+	fi
 fi
 AM_CONDITIONAL(HAVE_CAIRO, [test "x$HAVE_CAIRO" = xyes])
 
